### PR TITLE
Feature/tx construction

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-engine/src/main/java/com/radixdlt/atom/FixedTokenDefinition.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+package com.radixdlt.atom;
+
+import com.radixdlt.utils.UInt256;
+
+import java.util.Objects;
+
+/**
+ * Describes a fixed token definition
+ */
+public final class FixedTokenDefinition {
```
Comment (LukasGasior1) [inline radixdlt-engine/src/main/java/com/radixdlt/atom/FixedTokenDefinition.java:28]: does it need equals/hashCode?

File: radixdlt-engine/src/main/java/com/radixdlt/atom/SubstateId.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		this.substateId = Objects.requireNonNull(substateId);
+	}
+
+	public static SubstateId of(Particle particle) {
+		var dson = DefaultSerialization.getInstance().toDson(particle, DsonOutput.Output.ALL);
+		var substateHash = HashUtils.sha256(dson);
+		return new SubstateId(substateHash);
+	}
+
+	public static SubstateId ofVirtualSubstate(Particle particle) {
```
Comment (LukasGasior1) [inline radixdlt-engine/src/main/java/com/radixdlt/atom/SubstateId.java:45]: is this needed? seems the same as `of(Particle particle)`

Comment (talekhinezh) [reply radixdlt-engine/src/main/java/com/radixdlt/atom/SubstateId.java:45]: Just really a placeholder for now, they will be different pretty soon

**Review by LukasGasior1 — APPROVED** [review]

**Pull request merged**